### DISCUSSION TITLE
Generate templates and pipelines with AI

### DISCUSSION THREAD

**Discussion by kopetan-ms** [Azure/azure-dev#1768] [Ideas]
With the boom of generative AI (OpenAI and the bunch), it would be really cool if the Azure Dev Cli can allow me to describe the setup in human language like "I want to start with a asp.net web api, using dotnet 7, be wired up with App Insights and talk to CosmosDB using the Mongo API, built and deployed by Github. Also, please deploy this to Container Apps and scaffold the testing infrastructure".
Some of this can be already generated with ChatGPT, but perhaps Dev Cli can be even more perscriptive.

**Comment by gkaleta** [comment]
+1 co-pilot does help a bit down the way, but going a step further would be even cooler.

**Comment by ruslany** [comment]
+1 - it would be great if I could do this:

azd initwithai "create a containerapps environment with 2 containerapps and one container job with placeholder containerimages and with user assigned identities"

and that would generate all the bicep tempates in the infra folder without touching the src folder.

**Comment by savannahostrowski** [comment]
Thanks for the discussion!

We are currently working on two variants of features related to this topic:
- https://github.com/Azure/azure-dev/issues/705 - Right now, we are in the prototyping and evaluation stage of both AI/non-AI powered variants of a feature which enables a user to bring any repo and generates the "right" IaC.

 - https://github.com/Azure/azure-dev/issues/2342 - which is more tailored to a developer that has specific service combinations in mind

**Comment by savannahostrowski** [comment]
Oh and we also have an issue tracking generating CI/CD pipelines as well https://github.com/Azure/azure-dev/issues/2343
